app/controllers/tags_controller.rb

```diff
@@ -0,0 +1,50 @@
+class TagsController < ApplicationController
+  def index
+  end
+  def show
+  end
+  def new
+  end
+  #creates a new tag, also creates a category here if it doesn;t exist
+  def create
+    #uses this boolean to check for a tag already existing.  I'm sure there's way better
+    #methods for this
+    bool = "true"
```

Comment:
we could use `.find_or_create_by` to dry up this code: http://apidock.com/rails/v4.0.2/ActiveRecord/Relation/find_or_create_by